src/components/Auth/SignIn/SignIn.js

```diff
@@ -1,64 +1,73 @@
-    displayName: '',
-    email: '',
-    password: '',
-    confirmPassword: '',
-  });
-
-  const {email, password} = userCredentials;
-
-  const handleSubmit = async (event) => {
-    event.preventDefault();
-
-    try {
-      setUserCredentials({...userCredentials, email: '', password: ''});
-      await auth.signInWithEmailAndPassword(email, password);
-    } catch (error) {
-      console.log(error);
-    }
-  };
+const SignInSchema = Yup.object().shape({
+    email: Yup.string()
+        .typeError('Proszę podać poprawny adres email')
+        .required('Pole obowiązkowe')
+        .email('Proszę podać poprawny adres email')
+        .min(6, 'Minimum 6 znaków'),
+    password: Yup.string()
+        .min(6, 'Minimum 6 znaków')
+        .max(40, 'Maxymalnie 30 znaków')
+        .required('Pole obowiązkowe'),
+});
 
-  const handleChange = (event) => {
-    const {name, value} = event.target;
-
-    setUserCredentials({...userCredentials, [name]: value});
-  };
-
-  return (
-    <div className='sign-in'>
-      <h2>It great to have you back on your next training session</h2>
-      <p>Sign in and let's do this</p>
-      <form onSubmit={handleSubmit}>
-        <Input
-          name='email'
-          type='email'
-          label='Email'
-          value={email}
-          onChange={handleChange}
-          required
-        />
-        <Input
-          name='password'
-          type='password'
-          label='Password'
-          value={password}
-          onChange={handleChange}
-          required
-        />
-        <div className='sign-in__buttons'>
-          <Button type='submit'>Zaloguj się</Button>
-          <Button onClick={signInWithGoogle}>Zaloguj z Google</Button>
+const SignIn = () => {
+    return (
+        <div className='sign-in'>
+            <h2>It great to have you back on your next training session</h2>
+            <p>Sign in and let's do this</p>
+            <Formik
+                initialValues={{
+                    email: '',
+                    password: ''
+                }}
+                validationSchema={SignInSchema}
+                onSubmit={async (values, {setSubmitting}) => {
+                    try {
+                        await auth.signInWithEmailAndPassword(values.email, values.password);
+                    } catch (error) {
+                        console.log(error);
+                    }
+                    setSubmitting(false);
+                }}
+            >
+                {({
+                      values, handleChange,
+                  }) => (
+                    <Form>
+                        <Field
+                            label="E-mail"
+                            id="emailId"
+                            name="email"
+                            type="email"
+                            component={Input}
+                            value={values.email}
+                            onChange={handleChange}
```

Comment:
gdzie ta metoda handleChange?

Reply:
ona jest default-owa :

handleChange: (e: React.ChangeEvent<any>) => void
General input change event handler. This will update the values[key] where key is the event-emitting input's name attribute. If the name attribute is not present, handleChange will look for an input's id attribute. Note: "input" here means all HTML inputs.